shm/Ship.cpp

```diff
@@ -0,0 +1,36 @@
+#include <string>
+
+class Ship {
+public:
+    Ship()
+        : id_(-1) {}
+    Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id)
+        : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id) {}
+    Ship(int maxCrew, int speed, size_t id)
+        : Ship(0, maxCrew, speed, "", id) {}
+
+    void setName(const std::string& name) { name_ = name; }
+
+    Ship& operator-=(size_t num) {
+        crew_ -= num;
+        return *this;
+    }
+    Ship& operator+=(size_t num) {
+        crew_ += num;
+        return *this;
```

Comment:
How are checking if you have space in the ship you crewman you want to add?

Reply:
+1

Reply:
fixed